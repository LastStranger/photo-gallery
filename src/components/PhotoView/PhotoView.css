.photo-view {
  background-color: black;
}

.photo-viewer {
  height: 100vh;
  background-color: black;
  display: flex;
  justify-content: center;
  align-items: center;
}

.photo-view-img {
  max-width: 100%;
  max-height: 100vh;
}
