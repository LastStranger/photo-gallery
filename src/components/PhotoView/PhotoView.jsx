// @flow

import { h, Component } from 'preact';
import Header from '../Header';
import Navbar from '../Navbar';
import './PhotoView.css';

const MAX_MENU_VISIBILE = 5000;

class PhotoView extends Component {
  state = {
    visibileNav: false,
    photo: {}
  };
  componentDidMount() {
    fetch(`/flickr/photo/${this.props.id}`)
      .then(response => response.json())
      .then(flickrData => {
        this.setState({ photo: flickrData });
        this.handleVisibilityChange();
      })
      .catch(err => {
        console.error('error', err); // eslint-disable-line no-console
      });

    this.props.setLoading(true);
  }
  componentWillUnmount() {
    this.clearVisibilityTimeout();
    this.props.setLoading(true);
  }
  props: {
    id: string,
    setLoading: Function
  };
  visibileTimer = null;
  clearVisibilityTimeout() {
    if (this.visibileTimer) {
      clearTimeout(this.visibileTimer);
      this.visibileTimer = null;
    }
  }
  imgLoaded = () => {
    this.props.setLoading(false);
  };
  startLoading = () => {
    this.props.setLoading(true);
  };
  handleVisibilityChange = () => {
    this.clearVisibilityTimeout();

    const nowVisibile = !this.state.visibileNav;
    if (nowVisibile) {
      this.visibileTimer = setTimeout(() => this.setState({ visibileNav: !nowVisibile }), MAX_MENU_VISIBILE);
    }
    this.setState({ visibileNav: nowVisibile });
  };
  render() {
    return (
      <section
        aria-checked={this.state.visibileNav}
        role="switch"
        tabIndex="0"
        onClick={this.handleVisibilityChange}
        className="photo-view"
      >
        <Header
          startLoading={this.startLoading}
          visibile={this.state.visibileNav}
          invertColor
          showBackLink
          title={this.state.photo.title || ''}
        />
        <div className="photo-viewer">
          {this.state.photo.large
            ? <img
                onLoad={this.imgLoaded}
                className="photo-view-img"
                src={this.state.photo.large}
                alt={this.state.photo.title}
              />
            : null}
        </div>
        <Navbar visibile={this.state.visibileNav} />
      </section>
    );
  }
}

export default PhotoView;
