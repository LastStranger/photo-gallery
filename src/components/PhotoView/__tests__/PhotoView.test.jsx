// @flow

import { h } from 'preact';
import render from 'preact-render-to-string';
import PhotoView from '../';
import { photo } from '../../../testHelpers';

test('renders', () => {
  const string = render(<PhotoView id={photo.id} setLoading={() => {}} />);
  expect(string).toMatchSnapshot();
});

test('makes the correct ajax call', () => {
  global.fetch = jest.fn().mockImplementation(() => Promise.resolve({ json: () => {} }));
  const instance = new PhotoView();
  instance.props = { id: photo.id, setLoading: () => {} };
  instance.componentDidMount();
  expect(global.fetch).toBeCalledWith(`/flickr/photo/${photo.id}`);
});
