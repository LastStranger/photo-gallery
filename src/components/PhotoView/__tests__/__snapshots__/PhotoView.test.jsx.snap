// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<section role="switch" tabIndex="0" class="photo-view">
  <div class="header-container is-invisibile">
    <header class="header-bar header-bar-inverse">
      <h1 class="header-title"></h1>
    </header>
  </div>
  <div class="photo-viewer"></div>
  <nav class="navbar is-invisibile">
    <img class="navbar-item" src="/assets/icons/plusOne.jpg" alt="give +1"
    />
    <img class="navbar-item" src="/assets/icons/comment.jpg" alt="comment"
    />
    <img class="navbar-item" src="/assets/icons/add.jpg" alt="add" />
    <img class="navbar-item" src="/assets/icons/arrow.jpg" alt="share" />
  </nav>
</section>
`;
